### PR TITLE
Brave url scheme android

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Addresses https://github.com/brave/browser-android-tabs/issues/850.

Contains adaptation of PRs https://github.com/brave/brave-core/pull/584 and https://github.com/brave/brave-core/pull/164 for re-use with Android.

Requires PR from Android https://github.com/brave/browser-android-tabs/pull/912 .

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
  - [x] Android
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. Build and install apk.
2. Type in address bar `brave://version` .
3. Should be loaded originally known `chrome://version` page with `brave://version` in an url bar.
4. Repete 2-3 with `chrome://version`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: android/chromium_src/components/url_formatter/url_fixer.cc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+namespace {
+
+const char kBraveUIScheme[] = "brave";
+
+bool IsEquivalentScheme_BraveImpl(const std::string& scheme1,
+                                  const std::string& scheme2) {
+  return (scheme1 == kBraveUIScheme && scheme2 == kChromeUIScheme) ||
+         (scheme1 == kChromeUIScheme && scheme2 == kBraveUIScheme) ||
+         (scheme1 == url::kAboutScheme && scheme2 == kBraveUIScheme) ||
+         (scheme1 == kBraveUIScheme && scheme2 == url::kAboutScheme);
```
Comment (bbondy) [inline android/chromium_src/components/url_formatter/url_fixer.cc]: I think maybe you need to compare about and chrome schemes here?

**Comment by AlexeyBarabash** [comment]
@bbondy 
Yes, it works with `brave://about` and `about:version` urls.

This PR is spread over the two repos,  and it is also using `browser_about_handler.cc` at 
https://github.com/brave/browser-android-tabs/pull/912/files#diff-8b62ac59d92e56ecc354070fc66774aeR155
.

File: android/chromium_src/components/url_formatter/url_fixer.cc (hunk 1 above)
Comment (AlexeyBarabash) [reply android/chromium_src/components/url_formatter/url_fixer.cc]: Yes, I need this also, as I need to compare for equality all the pairs from `<brave, chrome, about>`:
```
brave<->chrome 
brave<->about
chrome<->about
```
But `chrome<->about` comparison is already done in IsEquivalentScheme_ChromiumImpl , https://cs.chromium.org/chromium/src/components/url_formatter/url_fixer.cc?type=cs&q=IsEquivalentScheme&g=0&l=683 
```
bool IsEquivalentScheme(const std::string& scheme1,
                        const std::string& scheme2) {
  return scheme1 == scheme2 ||
         (scheme1 == url::kAboutScheme && scheme2 == kChromeUIScheme) ||
         (scheme1 == kChromeUIScheme && scheme2 == url::kAboutScheme);
}
``` 
So I used compare `chrome<->about` by invoking `IsEquivalentScheme_ChromiumImp` .

File: components/toolbar/brave_toolbar_model_impl.h [hunk 2]
```diff
@@ -10,7 +10,13 @@
 class BraveToolbarModelImpl : public ToolbarModelImpl {
   public:
     using ToolbarModelImpl::ToolbarModelImpl;
+#if defined(OS_ANDROID)
+    // On Android GetURLForDisplay is not invoked as on brave-coer,
```
Comment (petemill) [inline components/toolbar/brave_toolbar_model_impl.h]: nit: spelling of brave-core

Comment (AlexeyBarabash) [reply components/toolbar/brave_toolbar_model_impl.h]: Thanks, fixed.

**Comment by bsclifton** [comment]
@AlexeyBarabash is this something that we'd still like to do? or can it be closed?

**Comment by AlexeyBarabash** [comment]
@bsclifton I will check if `brave://`works on current Android-core build.
If it already works, then will close this PR.
Otherwise either close this PR or update.

**Comment by AlexeyBarabash** [comment]
@bsclifton 
Just finished build and I see `brave://xxx` works, but when the page gets loaded, I see `chrome://xxx` in url bar. 
So the changes inherited from brave-core codebase work particularly.
I had met such situation while working on this PR and I will check which of these commits does fix of described about case.

**Comment by bsclifton** [comment]
Closing as stale - let's create a new PR if we address this in the future 😄

**Pull request closed without merging**